### PR TITLE
[DO NOT MERGE] Adding load table from JSON; fixing load from CSV example

### PR DESCRIPTION
**Pull request by telpirion**
This PR makes the following changes:
- Moves snippets out from BigqueryTest project to BigquerySample project (Snippets.cs).
- Adds bigquery_load_table_gcs_json sample.
- Adds BigqueryTest.TestImportDataFromJSON() test.
- Changes bigquery_load_table_gcs_csv sample.
- Changes BigqueryTest.TestImportFromCloudStorage() test.

### CODE REVIEW

**Comment by telpirion** [comment]
@beccca for the review. Thank you!

File: bigquery/api/BigquerySample/Snippets.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        /// Loads the table from a JSON source.
+        /// </summary>
+        /// <param name="client">BigQuery client</param>
+		/// <param name="datasetId"></param>
+		/// <param name="tableId"></param>
+		/// <param name="gcsURI"></param>
+		/// <param name="field1"></param>
+		/// <param name="field2"></param>
+        public static void LoadTableFromJSON(BigQueryClient client,
+		    string datasetId, string tableId, string gcsURI, string field1,
```
Comment (tswast) [inline bigquery/api/BigquerySample/Snippets.cs]: Could you use the concrete URI in the sample itself instead of just via the test?

File: bigquery/api/BigquerySample/Program.cs [hunk 2]
```diff
@@ -62,7 +62,7 @@ private static void Main(string[] args)
                 foreach (var row in result)
                 {
                     Console.WriteLine($"{row["url"]}: {row["view_count"]} views");
-                }
+                }
```
Comment (beccca) [inline bigquery/api/BigquerySample/Program.cs]: Watch out for things like this (extraneous whitespace was added, which shouldn't be there & shouldn't be part of this commit)

**Pro Tip:** _Before sending a Pull Request, scroll through the diff to double check correctness_

^--- I also do this locally before every commit, quick `git diff` locally to review changes I'm about to commit

File: bigquery/api/BigqueryTest/BigqueryTest.cs [hunk 3]
```diff
@@ -478,6 +455,35 @@ public void TestImportDataFromFile()
             Assert.Equal(uploadTestWordValue, row["unique_words"]);
         }
 
+        [Fact]
+        public void TestImportDataFromJSON()
+		{
```
Comment (beccca) [inline bigquery/api/BigqueryTest/BigqueryTest.cs]: Weird things going on with indentations/spaces in this diff. Double check the indentation of all of the code, overall. This whole test method is indented too much.

Comment (telpirion) [reply bigquery/api/BigqueryTest/BigqueryTest.cs]: Done. I hadn't yet configured VS Community to convert tabs to spaces.

**Comment by telpirion** [comment]
Addressed comments; pushed new commit.

@beccca, @tswast for follow-up review.

File: bigquery/api/BigquerySample/Snippets.cs [hunk 4]
```diff
@@ -48,33 +45,32 @@ public static class BiqQuerySnippets
...
-        /// <param name="field1">Field in the source file.</param>
-        /// <param name="field2">Another field in the source file.</param>
         public static void LoadTableFromCSV(string datasetId,
-		    string tableId, BigQueryClient client, string gcsURI, string field1, string field2)
+            string tableId, BigQueryClient client)
         {
-			// [START bigquery_load_table_gcs_csv]
-			var dataset = client.CreateDataset(datasetId);
+            // [START bigquery_load_table_gcs_csv]
+            var gcsURI = "gs://cloud-samples-data/bigquery/us-states/us-states.json";
```
Comment (tswast) [inline bigquery/api/BigquerySample/Snippets.cs]: Should be the CSV URL, right?

Comment (telpirion) [reply bigquery/api/BigquerySample/Snippets.cs]: Done.
